### PR TITLE
Add blog post about forensic container analysis

### PR DESCRIPTION
**Pull request by adrianreber**
This adds a blog article how to use to do forensic analysis on a container checkpoint.

https://github.com/kubernetes/enhancements/issues/2008

### CODE REVIEW

**Review by sftim — COMMENTED** [review]
Some early feedback - hope it's useful

File: content/en/blog/_posts/2023-03-10-forensic-container-analysis/index.md [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+
+This results in a container called `counter` running in a pod called `counters`.
+
+Once the container is running I am performing following actions with that
+container:
+
+```shell
+curl <container ip address>:8088/create?test-file
+curl <container ip address>:8088/secret?RANDOM_1432_KEY
+curl <container ip address>:8088
```
[the remark this answers is not in the document]
Comment (adrianreber) [reply content/en/blog/_posts/2023-03-10-forensic-container-analysis/index.md]: done

File: content/en/blog/_posts/2023-03-10-forensic-container-analysis/index.md [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+| CONTAINER |              IMAGE               |      ID      | RUNTIME |       CREATED       | ENGINE |     IP     | CHKPT SIZE | ROOT FS DIFF SIZE |
++-----------+----------------------------------+--------------+---------+---------------------+--------+------------+------------+-------------------+
+| counter   | quay.io/adrianreber/counter:blog | 059a219a22e5 | runc    | 2023-03-02T06:06:49 | CRI-O  | 10.88.0.23 | 8.6 MiB    | 3.0 KiB           |
++-----------+----------------------------------+--------------+---------+---------------------+--------+------------+------------+-------------------+
+CRIU dump statistics
++---------------+-------------+--------------+---------------+---------------+---------------+
+| FREEZING TIME | FROZEN TIME | MEMDUMP TIME | MEMWRITE TIME | PAGES SCANNED | PAGES WRITTEN |
++---------------+-------------+--------------+---------------+---------------+---------------+
+| 100809 us     | 119627 us   | 11602 us     | 7379 us       |          7800 |          2198 |
++---------------+-------------+--------------+---------------+---------------+---------------+
```
[the remark this answers is not in the document]
Comment (adrianreber) [reply content/en/blog/_posts/2023-03-10-forensic-container-analysis/index.md:106]: done

File: content/en/blog/_posts/2023-03-10-forensic-container-analysis/index.md [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+$ ls  checkpoint/pages-*
+checkpoint/pages-1.img  checkpoint/pages-2.img  checkpoint/pages-3.img
+```
+
+When I initially used the container I stored a random key (`RANDOM_1432_KEY`)
+somewhere in the memory. Let see if I can find it:
+
+```console
+$ grep -a RANDOM_1432_KEY checkpoint/pages-*
+checkpoint/pages-2.img:0ctione��e�� ft�����������textnt-type ft�����������TTP/1.0RANDOM_1432_KEY��e����t����������Content-type: text/html
```
[the remark this answers is not in the document]
Comment (adrianreber) [reply content/en/blog/_posts/2023-03-10-forensic-container-analysis/index.md]: changed

File: content/en/blog/_posts/2023-03-10-forensic-container-analysis/index.md [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+are just the starting point. Depending on your requirements it is possible to
+look at certain things in much more detail, but this article should give you an
+introduction how to start the analysis of your checkpoint.
+
+## How do I get involved?
+
+You can reach SIG Node by several means:
+
+* Slack: [#sig-node][slack-sig-node]
+* [Mailing list][sig-node-ml]
```
[the remark this answers is not in the document]
Comment (adrianreber) [reply content/en/blog/_posts/2023-03-10-forensic-container-analysis/index.md]: added

File: content/en/blog/_posts/2023-03-10-forensic-container-analysis/index.md [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+
+**Authors:** Adrian Reber (Red Hat)
+
+In my previous article, [Forensic container checkpointing in
+Kubernetes][forensic-blog], I introduced checkpointing in Kubernetes
+and how it has to be setup and how it can be used. The name of the
+feature is Forensic container checkpointing, but I did not go into
+any details how to do the actual analysis of the checkpoint created by
+Kubernetes. In this article I want to provide details how the
+checkpoint can be analyzed.
```
[the remark this answers is not in the document]
Comment (adrianreber) [reply content/en/blog/_posts/2023-03-10-forensic-container-analysis/index.md:16]: also added a sentence concerning alpha state

**Comment by adrianreber** [comment]
> Some early feedback - hope it's useful

Thanks. As always, very helpful.

**Comment by adrianreber** [comment]
@haircommander @saschagrunert please take a look, it is not as CRI-O related as the previous blog post, but as you also reviewed the previous blog post, maybe you also have some input for this PR. Thanks.

**Comment by haircommander** [comment]
LGTM, interesting blog and seems useful!

**Review by sftim — COMMENTED** [review]
Thanks
/lgtm
/approve

/hold
as there is feedback available.

If you want to @adrianreber you can open another PR that updates your previous article to link to this follow up.
(/hold that other PR until this one publishes)

**Comment by sftim** [comment]
@adrianreber you can unhold this PR right away, or you can do the wording tweak and get a new LGTM. Your choice.

**Comment by adrianreber** [comment]
> @adrianreber you can unhold this PR right away, or you can do the wording tweak and get a new LGTM. Your choice.

I added your suggestions and re-pushed.

**Comment by adrianreber** [comment]
> If you want to @adrianreber you can open another PR that updates your previous article to link to this follow up.
> (/hold that other PR until this one publishes)

#39842

**Review by saschagrunert — APPROVED** [review]
LGTM, great blog post!

**Review by sftim — COMMENTED** [review]
/lgtm
/hold cancel

**Pull request merged**